sdk/python/feast/online_response.py

```diff
@@ -94,25 +96,42 @@ def _infer_online_entity_rows(
     entity_rows_dicts = cast(List[Dict[str, Any]], entity_rows)
     entity_row_list = []
     entity_type_map = dict()
+    entity_value_map = defaultdict(list)
+
+    # Flatten keys-value dicts into lists for type inference
+    for entity in entity_rows_dicts:
+        for key, value in entity.items():
+            if isinstance(value, Value):
+                entity_type_map[key] = _proto_str_to_value_type(
```

Comment:
mind leaving a comment here on what's going on?

seems easy to miss that we use the last defined value as the type so if there are inconsistencies of values in entity rows it wouldn't be caught. But this is the old behavior too

Reply:
I just fixed the underlying issue 😉 (and added a comment).